src/output-lua.c

```diff
@@ -823,8 +825,8 @@ static OutputInitResult OutputLuaLogInit(ConfNode *conf)
         } else if (opts.alproto == ALPROTO_SSH) {
             om->TxLogFunc = LuaTxLogger;
             om->alproto = ALPROTO_SSH;
-            om->tc_log_progress = SshStateBannerDone;
-            om->ts_log_progress = SshStateBannerDone;
+            om->tc_log_progress = SshStateFinished;
```

Comment:
Why this change ?

Reply:
That is a good idea.
But then, there remains the question wether we lose the ability to use LUA for `ssh.proto_version` if we do not reach `SshStateKexDone` ? (for instance, just banner grabbing)

Reply:
My bad, if to have code like this:
```suggestion
            om->tc_log_progress = SshStateBannerDone;
            om->ts_log_progress = SshStateFinished;;
```
`ssh.proto_version` will be printed using lua for the corrupted ssh session. And hassh parameters will be printed for full version.

Reply:
@catenacyber @victorjulien I can add for now `if` for hassh and documentation that we can lose the ability to use LUA for ssh.proto_version if we do not reach `SshStateFinished`.
```suggestion
            if(rs_ssh_hassh_is_enabled())
                om->tc_log_progress = SshStateFinished;
            else
                om->tc_log_progress = SshStateBannerDone;
```

And I can add ticket for future to have smart `tc_log_progress` and `ts_log_progress` detection (callback function, which checks last state or ability to have few states at the same time)

Reply:
@catenacyber @victorjulien 
Also, if I have code like this:
```suggestion
            om->tc_log_progress = SshStateFinished;
            om->ts_log_progress = SshStateFinished;;
```
It works for incomplete (I tested session with only banner packets) ssh session and `ssh.proto.version` is shown.

Reply:
I wonder if its worth exploring use of `OutputModule::TxLogCondition` here. Than you can have a bit more advanced logic.

Reply:
Thanks for the pointer